tests/cnf/ran/internal/ranhelper/ranhelper.go

```diff
@@ -59,3 +62,13 @@ func ExecLocalCommand(timeout time.Duration, command string, args ...string) (st
 
 	return string(output), err
 }
+
+// GetPodNameWithLabel returns a pod name matching pod label selector in a given namespace.
+func GetPodNameWithLabel(client *clients.Settings, podNamespace, podLabelSelector string) (string, error) {
+	podList, err := pod.List(client, podNamespace, metav1.ListOptions{LabelSelector: podLabelSelector})
+
+	glog.V(ranparam.LogLevel).Infof("podList[0] matching podLabelSelector is '%v'",
+		podList[0].Definition.Name)
+
+	return podList[0].Definition.Name, err
```

Comment:
This will panic if no pods exist with the label or the list function returns an error

Reply:
Good catch. Thanks!
I have updated the function and checked it by running dry-run validation. It's passed on both scenarios - [1] pod existence and [2] pod non-existence cases.

**pod non-existence case validation logs:**
>   Validate detaching the AI multi-node openshift spoke cluster [75374, test_id:75374]
>   /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:77
>   > Enter [It] Validate detaching the AI multi-node openshift spoke cluster - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:77 @ 09/25/24 17:37:17.713
>   STEP: deleting default assisted installer template reference ConfigMap custom resources - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:88 @ 09/25/24 17:37:17.713
>   STEP: deleting default assisted installer cluster level templates ConfigMap CR - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:90 @ 09/25/24 17:37:17.713
>   STEP: deleting default assisted installer node level templates ConfigMap CR - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:99 @ 09/25/24 17:37:17.832
>   END STEP: deleting default assisted installer template reference ConfigMap custom resources - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:88 @ 09/25/24 17:37:17.968 (255ms)
>   STEP: deleting siteconfig operator pod running under rhacm namespace on hub cluster.  - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:129 @ 09/25/24 17:37:17.969
>   STEP: Get the siteconfig operator pod name with label app.kubernetes.io/name=siteconfig-controller - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:131 @ 09/25/24 17:37:17.969
>   END STEP: deleting siteconfig operator pod running under rhacm namespace on hub cluster.  - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:129 @ 09/25/24 17:37:18.093 (125ms)
>   [FAILED] Failed to get siteconfig operator pod name with label app.kubernetes.io/name=siteconfig-controller from rhacm namespace
>   Unexpected error:
>       <*errors.errorString | 0xc000d56270>: 
>       no pod found with label app.kubernetes.io/name=siteconfig-controller under namespace rhacm
>       {
>           s: "no pod found with label app.kubernetes.io/name=siteconfig-controller under namespace rhacm",
>       }
>   occurred
>   In [It] at: /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:134 @ 09/25/24 17:37:18.093
>   < Exit [It] Validate detaching the AI multi-node openshift spoke cluster - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:77 @ 09/25/24 17:37:18.093 (380ms)


**pod existence case validation logs:**
>  Validate detaching the AI multi-node openshift spoke cluster [75374, test_id:75374]
>   /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:77
>   > Enter [It] Validate detaching the AI multi-node openshift spoke cluster - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:77 @ 09/25/24 17:39:40.286
>   STEP: deleting default assisted installer template reference ConfigMap custom resources - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:88 @ 09/25/24 17:39:40.286
>   STEP: deleting default assisted installer cluster level templates ConfigMap CR - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:90 @ 09/25/24 17:39:40.286
>   STEP: deleting default assisted installer node level templates ConfigMap CR - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:99 @ 09/25/24 17:39:40.67
>   END STEP: deleting default assisted installer template reference ConfigMap custom resources - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:88 @ 09/25/24 17:39:41.206 (920ms)
>   STEP: deleting siteconfig operator pod running under rhacm namespace on hub cluster.  - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:129 @ 09/25/24 17:39:41.206
>   STEP: Get the siteconfig operator pod name with label app.kubernetes.io/name=siteconfig-controller - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:131 @ 09/25/24 17:39:41.206
> pod name%!(EXTRA string=siteconfig-controller-manager-665757d8d8-kfg7w)  STEP: Pull the siteconfig operator pod name from namespace rhacm - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:138 @ 09/25/24 17:39:41.413
>   END STEP: deleting siteconfig operator pod running under rhacm namespace on hub cluster.  - /home/pmohanra/peri_work/mpmaruthu/eco-gotests/tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go:129 @ 09/25/24 17:39:44.074 (2.868s)
> 

new patch reflecting the updated changes.